Documentation admin tables (#1560)
kartoza/sawps: docs/src/administrator/manual/django-admin.md

## Django Admin Tables

![Django Admin Tables](./img/admin-11.png)

1. **ACCOUNTS**: The administrators can use this section to perform operations related to accounts.

    - **Email Addresses**: The `Email addresses` table is designed for storing and managing the email addresses associated with user accounts. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing user email addresses.

2. **ACTIVITY**: The administrators can use this section to perform operations related to activities.

    - **Activities**: The `Activity` table is designed to capture and track various activities related to the management and conservation of wildlife or specific areas. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing activities.
        - **Add**: Click on the [add](django-add-data.md) to view how to add a new activity.

>         - **Edit**: Click on the [edit](django-change-data.md) to view how to edit an activity.

Reviewer: ```suggestion
        - **Edit**: Click [here](django-change-data.md) to view how to edit an activity.
```

3. **AUTHENTICATION AND AUTHORISATION**: The administrators can use this section to perform operations related to authentication and authorisation.

    - **Groups**: The `Groups` table is designed to organise and categorise users into distinct groups or roles within a web application. This table serves as a fundamental component of the application's access control and permission management system. It provides administrators with the capability to perform various operations, including creating, modifying, and viewing user groups.

>         - **Add**: Click on the [add](django-add-data.md) to view how to add a new group.

Reviewer: ```suggestion
        - **Add**: Click [here](django-add-data.md) to view how to add a new group.
```

>         - **Edit**: Click on the [edit](django-change-data.md) to view how to edit a group.

Reviewer: ```suggestion
        - **Edit**: Click [here](django-change-data.md) to view how to edit a group.
```

    - **Permissions**: The `Permissions` table is a core element of a web application's access control and authorisation system. It is designed to manage and regulate the rights and privileges of users within the application, allowing administrators to create, modify, and oversee user permissions.

    - **Users**: The `Users` table is a foundational component of a web application, designed to represent and manage individual user accounts. This table plays a central role in user authentication, identity management, and access control. It empowers administrators to perform a range of essential operations, including user creation, modification, and access management.

>         - **Add**: Click on the [add](django-add-data.md) to view how to add a new user.

Reviewer: ```suggestion
        - **Add**: Click [here](django-add-data.md) to view how to add a new user.
```

>         - **Edit**: Click on the [edit](django-change-data.md) to view how to edit a user.

Reviewer: ```suggestion
        - **Edit**: Click [here](django-change-data.md) to view how to edit a user.
```

4. **CELERY RESULTS**: The administrators can use this section to perform operations related to celery.

    - **Group Results**: In the `Group result` table, administrators can click on a specific group to see details about its tasks and their outcomes. This is useful for tracking the progress of related tasks and ensuring they were executed successfully. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing group results.

    - **Task Results**: In the `Task result` table, administrators can view the status and outcomes of individual tasks. This is particularly valuable for diagnosing issues with specific tasks and understanding how the application is handling asynchronous processes. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing task results.

![Django Admin Tables](./img/admin-12.png)

> 5. **DOCUMENTATION CENTER**: The administrators can use this section to perform operations related to documentation. The `Documentation Center` is a vital tool for administrators to streamline the process of providing users with access to relevant documentation, guides, and help resources. It offers a structured approach to managing and presenting documentation, enhancing the user experience and support for SAWPS.

Reviewer: ```suggestion
5. **DOCUMENTATION CENTRE**: The administrators can use this section to perform operations related to documentation. The `Documentation Centre` is a vital tool for administrators to streamline the process of providing users with access to relevant documentation, guides, and help resources. It offers a structured approach to managing and presenting documentation, enhancing the user experience and support for SAWPS.
```

    - **Blocks**: In the `Blocks` table, administrators can specify a relative URL that links to the documentation base URL. This URL is used to auto-fetch content, allowing administrators to seamlessly integrate the documentation with SAWPS. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing blocks.

    - **Pages**: In the `Pages` table, administrators can specify the page with a relative URL that links to the documentation base URL. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing pages.

>     - **Preferences**: In the `Preferences` table, administrators can set the documentation base URL. This is the foundation of your documentation center and should be carefully defined to ensure accurate integration. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing preferences.

Reviewer: ```suggestion
    - **Preferences**: In the `Preferences` table, administrators can set the documentation base URL. This is the foundation of the documentation centre and should be carefully defined to ensure accurate integration. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing preferences.
```

6. **EASY AUDIT APPLICATION**: The administrators can use this section to perform operations related to the easy audit application. An `Easy Audit Application` is a tool that helps administrators monitor and maintain an audit trail of changes and user activity within the application, providing transparency, accountability, and security.

    - **CRUD events**: In the `CRUD events` table, administrators can track CRUD events to maintain data integrity, improve accountability, and ensure compliance with regulatory requirements. An audit log of these events helps administrators to keep a record of data changes, and understand who made the changes, and when they occurred. This table provides administrators with the capability to perform various operations, including purging, deleting, updating, and viewing CRUD events.

>     - **Login events**: In the `Login events` table, administrators can monitor login events which is critical for maintaining the security of SAWPS and keeping track of who is accessing the system. It helps in identifying suspicious or unauthorised activities, improving accountability, and enhancing the overall security posture of your application. This table provides administrators with the capability to perform various operations, including purging, deleting, updating, and viewing login events.

Reviewer: ```suggestion
    - **Login events**: In the `Login events` table, administrators can monitor login events which is critical for maintaining the security of SAWPS and keeping track of who is accessing the system. It helps in identifying suspicious or unauthorised activities, improving accountability, and enhancing the overall security posture of the application. This table provides administrators with the capability to perform various operations, including purging, deleting, updating, and viewing login events.
```

![Django Admin Tables](./img/admin-13.png)

8. **NOTIFICATION**: The administrators can use this section to perform operations related to reminders.

    - **Reminders**: The `Reminders` table is used for storing the reminder records. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing reminders.

9. **OCCURRENCE**: The administrators can use this section to perform operations related to occurrences.

    - **Survey methods**: The `Survey methods` table is used for storing the survey methods records. In the field of wildlife and environmental studies, survey methods are essential for systematically collecting data about the occurrence and distribution of various species and ecosystems. These methods are carefully designed to provide accurate information for research, conservation, and management purposes. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing survey methods.

> 10. **OTP STATIC**: The administrators can use this section to perform operations related to static OTP devices.

Reviewer: Not every one knows what the acronym stands for so explaining it once is required.
```suggestion
10. **OTP STATIC**: The administrators can use this section to perform operations related to static OTP (one-time pin) devices.
```

Author: okay

    - **Static devices**: The `Static devices` table is used for storing the static devices records. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing static devices.

> 11. **OTP TOTP**: The administrators can use this section to perform operations related to OTP TOTP devices.

Reviewer: ```suggestion
11. **OTP TOTP**: The administrators can use this section to perform operations related to OTP TOTP (time-based one-time pin) devices.
```

19. **SPECIES**: The administrators can use this section to perform operations related to species.

    - **Owned Species**: The `Owned Species` table is used for storing the owned species records. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing owned species.

    - **Taxa**: The `Taxa` table is used for storing the taxon(species) records. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing taxon(species).

    - **Taxon Ranks**: The `Taxon Ranks` table is used for storing the taxon rank records. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing taxon ranks.

> 20. - **STAKEHOLDER**: The administrators can use this section to perform operations related to stakeholders.

Reviewer: ```suggestion
20. **STAKEHOLDER**: The administrators can use this section to perform operations related to stakeholders.
```

    - **Reminders**: The `Reminders` table is used for storing the reminder records. The administrators can use the reminders table to create and manage reminders within the SAWPS. Reminders help users keep track of important tasks and deadlines. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing reminders.

    - **Titles**: The `Titles` table is used for storing the titles records. The administrators can use the titles table to manage and add titles that can be associated with user profiles within the SAWPS. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing titles.

    - **User roles**: The `User roles` table is used for storing the user roles records. User roles define the permissions and access levels for different users within the SAWPS. By adding user roles, administrators can control what users can and cannot do in the system. This table provides administrators with the capability to perform various operations, including adding, deleting, updating, and viewing user roles.

> * For more information, click on the [table](django-table.md) to view the table's documentation.

Reviewer: ```suggestion
* For more information, click [here](django-table.md) to view the table documentation.
```

> * For more information, click on the [add](django-add-data.md) to view how to add new data.

Reviewer: ```suggestion
* For more information, click [here](django-add-data.md) to view how to add new data.
```
